Source/Charts/Renderers/PieChartRenderer.swift

```diff
@@ -496,14 +496,27 @@ open class PieChartRenderer: NSObject, DataRenderer
                         }
                     }
                     else if drawYOutside
+                        
                     {
-                        context.drawText(valueText,
-                                         at: CGPoint(x: labelPoint.x,
-                                                     y: labelPoint.y + lineHeight / 2.0),
-                                         align: align,
-                                         angleRadians: angleRadians,
-                                         attributes: [.font: valueFont,
-                                                      .foregroundColor: valueTextColor])
+                        if let image = pe?.image {
```

Comment:
Rather than call long functions twice, declare what's different and call the function once. i.e.:
```
let textOrigin: CGPoint
if let image = pe?.image {
    textOrigin = X
} else {
    textOrigin = Y
}
context.drawText(....)
```